server/datastore/mysql/apple_mdm.go

```diff
@@ -1125,28 +1126,58 @@ func upsertMDMAppleHostLabelMembershipDB(ctx context.Context, tx sqlx.ExtContext
 	return nil
 }
 
-func (ds *Datastore) deleteMDMAppleProfilesForHost(ctx context.Context, tx sqlx.ExtContext, uuid string) error {
-	_, err := tx.ExecContext(ctx, `
-                    DELETE FROM host_mdm_apple_profiles
-                    WHERE host_uuid = ?`, uuid)
-	if err != nil {
-		return ctxerr.Wrap(ctx, err, "removing all profiles from host")
+// deleteMDMOSCustomSettingsForHost deletes configuration profiles and
+// declarations for a host based on its platform.
+func (ds *Datastore) deleteMDMOSCustomSettingsForHost(ctx context.Context, tx sqlx.ExtContext, uuid, platform string) error {
+
+	tableMap := map[string][]string{
+		"darwin":  {"host_mdm_apple_profiles", "host_mdm_apple_declarations"},
+		"windows": {"host_mdm_windows_profiles"},
+	}
+
+	tables, ok := tableMap[platform]
+	if !ok {
+		return ctxerr.Errorf(ctx, "unsupported platform %s", platform)
+	}
+
+	for _, table := range tables {
+		_, err := tx.ExecContext(ctx, fmt.Sprintf(`
+                    DELETE FROM %s
+                    WHERE host_uuid = ?`, table), uuid)
+		if err != nil {
+			return ctxerr.Wrapf(ctx, err, "removing all %s from host %s", table, uuid)
+		}
 	}
+
 	return nil
 }
 
-func (ds *Datastore) UpdateHostTablesOnMDMUnenroll(ctx context.Context, uuid string) error {
-	return ds.withTx(ctx, func(tx sqlx.ExtContext) error {
-		var hostID uint
-		row := tx.QueryRowxContext(ctx, `SELECT id FROM hosts WHERE uuid = ?`, uuid)
-		err := row.Scan(&hostID)
+func (ds *Datastore) MDMTurnOff(ctx context.Context, uuid string) error {
```

Comment:
Not required for this PR, but for a future refactor we should probably move those now-platform-agnostic methods out of `apple_mdm.go` and into `mdm.go`.